lib/core_extensions.rb

```diff
@@ -38,10 +38,20 @@ def initialize(*attribute)
     end
 
     def before_destroy(record)
-      klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
-          record.errors.add :base, _("%{record} is used by %{what}") % { :record => record, :what => what }
+      klasses.each do |klass, klass_name = klass|
+        record.association(klass.to_sym).association_scope.each do |what|
+          error_message = _("%{record} is used by %{what}")
+          unless what.is_a? String
+            authorized_associations = AssociationAuthorizer.authorized_associations(record.class.reflect_on_association(klass.to_sym).klass, klass_name, false)
+            if (!authorized_associations.is_a?(Array) || authorized_associations.include?(what)) &&
```

Comment:
I think this could do with test coverage - I tried to re-test it based solely on authorisations, and it still showed me the host name when deleting the host group.  It appears that authorized_associations is a Host::Managed::FriendlyIdActiveRecordRelation (!) and not an Array.